From b94e00d420084eb0626d2ab276882fff457e8867 Mon Sep 17 00:00:00 2001
From: Sebastian Lenartowicz <Sebastian.Lenartowicz@windriver.com>
Date: Fri, 13 Dec 2013 08:59:04 -0500
Subject: [PATCH 1/1] Adds some basic argument checking

This patch adds some basic argument checking to the "example" tools
(some of which are sufficiently useful to merit being cleaned up in
this manner), in a manner consistent with the tools that do already
have it (a few of the newer ones do).

Signed-off-by: Sebastian Lenartowicz <Sebastian.Lenartowicz@windriver.com>
---
 examples/dump-memory.c              |    5 +++++
 examples/map-addr.c                 |    5 +++++
 examples/map-symbol.c               |    6 ++++++
 examples/module-list.c              |    5 +++++
 examples/msr-event-example.c        |    2 +-
 examples/process-list.c             |    9 ++++-----
 examples/singlestep-event-example.c |    2 +-
 examples/win-guid.c                 |    9 ++++-----
 8 files changed, 31 insertions(+), 12 deletions(-)

diff --git a/examples/dump-memory.c b/examples/dump-memory.c
index a7d21e6..2c2cb81 100644
--- a/examples/dump-memory.c
+++ b/examples/dump-memory.c
@@ -48,6 +48,11 @@ main(
     uint32_t offset = 0;
     addr_t address = 0;
 
+    if(argc != 3){
+        fprintf(stderr, "Usage: dump-memory <name of VM> <name of output file>\n");
+        exit(1);
+    }
+
     /* this is the VM or file that we are looking at */
     char *name = argv[1];
 
diff --git a/examples/map-addr.c b/examples/map-addr.c
index c82d98a..d60bb31 100644
--- a/examples/map-addr.c
+++ b/examples/map-addr.c
@@ -38,6 +38,11 @@ main(
     int argc,
     char **argv)
 {
+    if(argc != 3){
+        fprintf(stderr, "Usage: map-addr <name of VM> <memory address to examine>\n");
+        exit(1);
+    }
+
     vmi_instance_t vmi;
     unsigned char *memory = malloc(PAGE_SIZE);
 
diff --git a/examples/map-symbol.c b/examples/map-symbol.c
index bedafbe..dcd0ded 100644
--- a/examples/map-symbol.c
+++ b/examples/map-symbol.c
@@ -37,6 +37,12 @@ main(
     int argc,
     char **argv)
 {
+
+    if(argc != 3){
+        fprintf(stderr, "Usage: map-symbol <name of VM> <name of symbol>\n");
+        exit(1);
+    }
+
     vmi_instance_t vmi;
     unsigned char *memory = malloc(PAGE_SIZE);
 
diff --git a/examples/module-list.c b/examples/module-list.c
index 597127c..a47c662 100644
--- a/examples/module-list.c
+++ b/examples/module-list.c
@@ -40,6 +40,11 @@ main(
     uint32_t offset;
     addr_t next_module, list_head;
 
+    if(argc != 2){
+        fprintf(stderr, "Usage: list-modules <name of VM>\n");
+        exit(1);
+    }
+
     /* this is the VM or file that we are looking at */
     char *name = argv[1];
 
diff --git a/examples/msr-event-example.c b/examples/msr-event-example.c
index cc188ad..81e99e6 100644
--- a/examples/msr-event-example.c
+++ b/examples/msr-event-example.c
@@ -54,7 +54,7 @@ int main (int argc, char **argv) {
 
     char *name = NULL;
 
-    if(argc < 2){
+    if(argc != 2){
         fprintf(stderr, "Usage: msr_events_example <name of VM>\n");
         exit(1);
     }
diff --git a/examples/process-list.c b/examples/process-list.c
index 2d2ed46..095b97b 100644
--- a/examples/process-list.c
+++ b/examples/process-list.c
@@ -44,11 +44,10 @@ int main (int argc, char **argv)
     unsigned long tasks_offset, pid_offset, name_offset;
     status_t status;
 
-    /* this is the VM or file that we are looking at */
-    if (argc != 2) {
-        printf("Usage: %s <vmname>\n", argv[0]);
-        return 1;
-    } // if
+    if(argc != 2){
+        fprintf(stderr, "Usage: process-list <name of VM>\n");
+        exit(1);
+    }
 
     char *name = argv[1];
 
diff --git a/examples/singlestep-event-example.c b/examples/singlestep-event-example.c
index 94945f0..689b1ea 100644
--- a/examples/singlestep-event-example.c
+++ b/examples/singlestep-event-example.c
@@ -55,7 +55,7 @@ int main (int argc, char **argv) {
 
     char *name = NULL;
 
-    if(argc < 2){
+    if(argc != 2){
         fprintf(stderr, "Usage: single_step_example <name of VM> \n");
         exit(1);
     }
diff --git a/examples/win-guid.c b/examples/win-guid.c
index 5c13c80..3c9e25e 100644
--- a/examples/win-guid.c
+++ b/examples/win-guid.c
@@ -244,11 +244,10 @@ int main(int argc, char **argv) {
 
     vmi_instance_t vmi;
 
-    /* this is the VM that we are looking at */
-    if (argc != 3) {
-        printf("Usage: %s name|domid <domain name|domain id>\n", argv[0]);
-        return 1;
-    }   // if
+    if(argc != 3) {
+        fprintf(stderr, "Usage: events_example name|domid <domain name|domain id>\n");
+        exit(1);
+    }
 
     uint32_t domid = VMI_INVALID_DOMID;
     GHashTable *config = g_hash_table_new(g_str_hash, g_str_equal);
-- 
1.7.9.7

